lib/src/view/analysis/tree_view.dart

```diff
@@ -162,124 +131,654 @@ class _InlineTreeViewState extends ConsumerState<AnalysisTreeView> {
+  );
+
+  final sidelineNodes = [firstNode, ...firstNode.mainline];
+
+  var path = initialPath;
+  return [
+    if (followsComment) const WidgetSpan(child: SizedBox(width: 4.0)),
+    ...sidelineNodes.mapIndexedAndLast(
+      (i, node, last) {
+        final pathToNode = path;
+        path = path + node.id;
+
+        return [
+          if (i == 0) ...[
+            if (followsComment) const WidgetSpan(child: SizedBox(width: 4.0)),
+            TextSpan(
+              text: '(',
+              style: textStyle,
+            ),
+          ],
+          ..._moveWithComment(
+            node,
+            parent: i == 0 ? parent : sidelineNodes[i - 1],
+            lineInfo: (
+              type: _LineType.inlineSideline,
+              startLine: i == 0 || sidelineNodes[i - 1].hasTextComment
+            ),
+            pathToNode: pathToNode,
+            textStyle: textStyle,
+            params: params,
+          ),
+          if (last)
+            TextSpan(
+              text: ')',
+              style: textStyle,
+            ),
+        ];
+      },
+    ).flattened,
+    const WidgetSpan(child: SizedBox(width: 4.0)),
+  ];
+}
+
+const _baseTextStyle = TextStyle(
+  fontSize: 16.0,
+  height: 1.5,
+);
+
+/// The different types of lines (move sequences) that are displayed in the tree view.
+enum _LineType {
+  /// (A part of) the game's main line.
+  mainline,
+
+  /// A sideline branching off the main line or a parent sideline.
+  /// Each sideline is rendered on a new line and indented.
+  sideline,
+
+  /// A short sideline without any branching, displayed in parantheses inline with it's parent line.
+  inlineSideline,
+}
+
+/// Metadata about a move's role in the tree view.
+typedef _LineInfo = ({_LineType type, bool startLine});
+
+List<InlineSpan> _moveWithComment(
+  ViewBranch branch, {
+  required ViewNode parent,
+  required TextStyle textStyle,
+  required _LineInfo lineInfo,
+  required UciPath pathToNode,
+  required _PgnTreeViewParams params,
+  GlobalKey? moveKey,
+}) {
+  return [
+    WidgetSpan(
+      alignment: PlaceholderAlignment.middle,
+      child: InlineMove(
+        branch: branch,
         parent: parent,
-        branch: firstChild,
-        isCurrentMove: currentMove,
-        key: currentMove ? currentMoveKey : null,
-        shouldShowAnnotations: shouldShowAnnotations,
-        shouldShowComments: shouldShowComments,
-        isSideline: !inMainline,
-        startMainline: startMainline,
-        startSideline: startSideline,
-        endSideline: !inMainline && firstChild.children.isEmpty,
+        lineInfo: lineInfo,
+        path: pathToNode + branch.id,
+        key: moveKey,
+        textStyle: textStyle,
+        params: params,
       ),
+    ),
+    if (params.shouldShowComments && branch.hasTextComment)
+      ..._comments(branch.comments!, textStyle: textStyle),
+  ];
+}
+
+class _SideLinePart extends ConsumerWidget {
```

Comment:
I'd add a doc for this class to improve readability. What is it? a part of a sideline?